src/java/com/unifina/signalpath/custom/AbstractJavaCodeWrapper.java

```diff
@@ -171,22 +171,21 @@ private String combineCode(String className) {
 				long line = d.getLineNumber()- StringUtils.countMatches(makeImportString(), "\n")-StringUtils.countMatches(getHeader(), "\n");
 
 				sb.append("Line ");
-				sb.append(Long.toString(line));
+				sb.append(line);
 				sb.append(": ");
 				sb.append(d.getMessage(null));
 				sb.append("\n");
 
 				CompilationErrorMessage msg = new CompilationErrorMessage();
 				msg.addError(line, d.getMessage(null));
-				msgs.add(new ModuleExceptionMessage(hash,msg));
+				msgs.add(new JavaCompilerErrorMessage(hash, msg));
```

Comment:
`CompilationErrorMessage` seems unnecessary and we want to get rid of that `Map` passing anyway in favour of object-oriented. So you should just include the info you need in `JavaCompilerErrorMessage`:

`new JavaCompilerErrorMessage(hash, line, d.getMessage(null))`